crates/subspace-archiving/src/archiver.rs

```diff
@@ -185,6 +185,33 @@ pub struct NewArchivedSegment {
     pub object_mapping: Vec<PieceObjectMapping>,
 }
 
+impl NewArchivedSegment {
+    /// Returns all the object mappings in this archived segment.
+    //
+    // TODO: make this a double-ended lazy iterator, for efficiency
+    pub fn object_mappings(&self) -> Vec<ObjectMapping> {
+        // Iterate through the object mapping vector for each piece
+        self.object_mapping
+            .iter()
+            .enumerate()
+            .flat_map(|(piece_position, mappings)| {
+                // And then through each individual object mapping in the piece
+                mappings
+                    .objects
+                    .iter()
+                    .map(move |piece_object| ObjectMapping::V0 {
+                        object_id: piece_object.hash(),
+                        first_piece_index: PieceIndex::from_segment_and_position(
+                            self.segment_header.segment_index(),
+                            piece_position as u32,
+                        ),
+                        first_piece_offset: piece_object.offset(),
```

Comment:
How do I translate raw offsets to regular offsets? I had a look for an API, but couldn’t find anything.

Reply:
As described above, you just need to account for the fact for every 31 byte of the piece offset there is actually 32nd byte not accounted for. So if offset was 5 in the piece, it remains 5, but if offset was 50, then it is actually 51 in practice and archiver needs to be corrected to account for this since I don't think I simply forgotten to remove the TODO.

Not critical to do in this exact PR since it was like this for a long time, but once we get to the actual retrieval, it will be important to have correct offset or we'll not get original data back. Just something I was reminded of reading through the code (very handy that I left TODO for myself there, but not great that I didn't fix it yet).

Reply:
I applied your suggestions (and renames) in [9937e8e](https://github.com/autonomys/subspace/pull/2979/commits/9937e8ef064212add8c321825d766f9921561e04), and added a checklist item in the notion doc for the raw to record offset conversion.